### PR TITLE
Informal definition of quad.

### LINKED ISSUE #5: Define term "quad"
The term ***quad*** is used in a number of specifications (e.g., n-quads, rdf-canon, sparql), but never really defined. Even though quads are not part of the abstract syntax, at least an informative definition within [4. RDF Datasets](https://w3c.github.io/rdf-concepts/spec/#section-dataset) would be useful as a common reference point.

> A ***quad*** is the combination of an [RDF statement](https://w3c.github.io/rdf-concepts/spec/#dfn-statement) and optionally,  a [named graph](https://w3c.github.io/rdf-concepts/spec/#dfn-named-graph).

See https://github.com/w3c/rdf-canon/issues/62

### PR DESCRIPTION
**Pull request by gkellogg**
Fixes #5.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-concepts/pull/12.html" title="Last updated on Feb 5, 2023, 9:38 PM UTC (c28b23e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-concepts/12/6539b58...c28b23e.html" title="Last updated on Feb 5, 2023, 9:38 PM UTC (c28b23e)">Diff</a>

### CODE REVIEW

File: spec/index.html [hunk 1]
```diff
@@ -816,6 +816,27 @@ <h3>Content Negotiation of RDF Datasets</h3>
       the consumer is expected to use the <a data-lt="rdf dataset">RDF dataset's</a> default graph.</p>
 
     </section>
+
+    <section id="section-dataset-quad" class="informative">
+      <h3>Dataset as a set of quads</h3>
+
+      <p>A <dfn>quad</dfn> is a conceptual projection of a <a>triple</a> with
+        either the <a>default graph</a> or a <a>named graph</a>.
+        A <a>quad</a> is the generalization of a <a>triple</a>
```
Comment (afs) [inline spec/index.html:825]: This is quite different to #5.

The projection (as in "a view of" - c.f. SQL) is the other way round - a triple is the projection of a quad without the graph name component.

In #5, a quad is a triple and an optional graph name.

File: spec/index.html [hunk 2]
```diff
@@ -816,6 +816,27 @@ <h3>Content Negotiation of RDF Datasets</h3>
...
+        along with an optional <a>graph name</a>,
+        indicating a named graph containing that <a>triple</a>.
+      </p>
+
+      <p>A <a>quad</a> can be represented as a tuple composed of <a>subject</a>, <a>predicate</a>, <a>object</a>,
+        and an optional <a>graph name</a>.</p>
+
+      <p>An <a>RDF dataset</a> can informally be considered to be a set of <a>quads</a>.</p>
+
+      <p class="note">Although a <a>quad</a> in the <a>default graph</a> shares
```
Comment (afs) [inline spec/index.html:835]: A quad is not "in" a graph.  A graph is a set of triples.

The text should say how quads relate to the graph of a dataset.

If we follow #5 and a quad is "triple+graph name":
```
<p>An <a>RDF dataset</a> can be considered to be a set of <a>quads</a>
where the quads no graph name supply the triples of the default graph
and the quads with the same graph name, supply the triples of the named graph with
that name.</p>
```

File: spec/index.html [hunk 3]
```diff
@@ -816,6 +816,27 @@ <h3>Content Negotiation of RDF Datasets</h3>
...
+      <h3>Dataset as a set of quads</h3>
+
+      <p>A <dfn>quad</dfn> is a conceptual projection of a <a>triple</a> with
+        either the <a>default graph</a> or a <a>named graph</a>.
+        A <a>quad</a> is the generalization of a <a>triple</a>
+        along with an optional <a>graph name</a>,
+        indicating a named graph containing that <a>triple</a>.
+      </p>
+
+      <p>A <a>quad</a> can be represented as a tuple composed of <a>subject</a>, <a>predicate</a>, <a>object</a>,
```
Comment (afs) [inline spec/index.html:830]: That works because when the text says "subject" it has to be subject of something - the triple.

**Review by afs — CHANGES_REQUESTED** [review]

File: spec/index.html [hunk 4]
```diff
@@ -816,6 +816,27 @@ <h3>Content Negotiation of RDF Datasets</h3>
       the consumer is expected to use the <a data-lt="rdf dataset">RDF dataset's</a> default graph.</p>
 
     </section>
+
+    <section id="section-dataset-quad" class="informative">
+      <h3>Dataset as a set of quads</h3>
```
Comment (hartig) [inline spec/index.html:821]: Just to be consistent with the capitalization of the other section titles.
```suggestion
      <h3>Dataset as a Set of Quads</h3>
```

File: spec/index.html (hunk 1 above)
Comment (hartig) [reply spec/index.html:825]: I also find the phrasing of the first sentence not so suitable. It is not really clear what a "conceptual projection" is.

My definition would be:
```
A <dfn>quad</dfn> is a <a>triple</a> associated with an optional <a>graph name</a> and
is used when referring to triples within an <a>RDF dataset</a>.
```

Using that definition as the first paragraph of the section, as the next paragraph I would put the sentence about "An RDF dataset can be .." (as given in [Andy's comment below](https://github.com/w3c/rdf-concepts/pull/12/files#r1094292891)).

As third paragraph I would then use the following adaptation of the paragraph about quads versus triples.
```
Although a <a>quad</a> without a graph name consists of the same three components as
a <a>triple</a>, it is a distinct concept, as it specifically captures the notion of
a triple within the <a>default graph</a> of an <a>RDF dataset</a>.
```

**Review by hartig — CHANGES_REQUESTED** [review]

File: spec/index.html (hunk 2 above)
Comment (TallTed) [reply spec/index.html:835]: Punctuation and language tweaks, and addition of a missing word...
```
<p>An <a>RDF dataset</a> can be considered to be a set of <a>quads</a>
where quads with no graph name supply the triples of the default graph,
and quads with the same graph name supply the triples of the named graph with
that name.</p>
```

**Comment by gkellogg** [comment]
Thanks, the original was a bit rushed, as I had the dual goal of trying something for PR Preview, which seems to be fixed now.

I've applied the suggestions from above, although I did keep the second sentence, which I find useful. Of course, let me know if you feel otherwise.

      <p>A <a>quad</a> can be represented as a tuple composed of <a>subject</a>, <a>predicate</a>, <a>object</a>,
        and an optional <a>graph name</a>.</p>

**Review by hartig — APPROVED** [review]
Looks good to me now. Thanks!

**Review by hartig — APPROVED** [review]

**Review by afs — APPROVED** [review]

**Pull request merged**